### PR TITLE
media: i2c: Add IVSC handshake support

### PR DESCRIPTION
**Pull request by ccgjimlai**
Add IVSC handshake support for OV02E and HM2172

### CODE REVIEW

File: drivers/media/i2c/hm2172.c [hunk 1]
```diff
@@ -910,10 +921,20 @@ struct hm2172 {
 	struct v4l2_ctrl *hblank;
 	struct v4l2_ctrl *exposure;
 
-	struct clk *img_clk;
-	struct regulator *avdd;
+	/* GPIO for reset */
 	struct gpio_desc *reset;
+	/* GPIO for Lattice handshake */
 	struct gpio_desc *handshake;
+	/* regulator */
```
Comment (paulmenzel) [inline drivers/media/i2c/hm2172.c:928]: The comment does not add anything useful.

**Review by paulmenzel — COMMENTED** [review]
Thank you for your patch. It’d be great to have a more elaborate commit message.

File: drivers/media/i2c/hm2172.c (hunk 1 above)
Comment (ccgjimlai) [reply drivers/media/i2c/hm2172.c]: Removed

File: drivers/media/i2c/hm2172.c [hunk 2]
```diff
@@ -15,6 +15,17 @@
 #include <media/v4l2-fwnode.h>
 #include <linux/clk.h>
 #include <linux/gpio/consumer.h>
+#if IS_ENABLED(CONFIG_INTEL_VSC)
+#include <linux/vsc.h>
+
+static const struct acpi_device_id cvfd_ids[] = {
+	{ "INTC1059", 0 },
+	{ "INTC1095", 0 },
+	{ "INTC100A", 0 },
```
[the remark this answers is not in the document]
Comment (hao-yao) [reply drivers/media/i2c/hm2172.c:24]: This is actually sorted by hardware generation (Tiger/Alder/Raptor/Meteor Lake)

**Review by hao-yao — APPROVED** [review]

**Pull request merged**